### ISSUE TITLE
Add max concurrency support for AWS Lambda functions when using AWS SQS as an event source 

### ISSUE THREAD

**Issue by htxiong** [getlift/lift#294]
### Use-case description

12 Jan 2023 AWS introduced maximum concurrency of AWS Lambda functions when using Amazon SQS as an event source (see release [announcement here](https://aws.amazon.com/about-aws/whats-new/2023/01/aws-lambda-maximum-concurrency-amazon-sqs-event-source/) and [compute blog from AWS](https://aws.amazon.com/blogs/compute/introducing-maximum-concurrency-of-aws-lambda-functions-when-using-amazon-sqs-as-an-event-source/))

And Serverless started to support it around 3 weeks. [PR](https://github.com/serverless/serverless/issues/11676)

I suggest we should define max concurrency in Lift.

### Example Config

```
    alarm: engineering@xxx.com
    batchSize: 1
    maxRetries: 5
    maxConcurrency: 10
    worker:
```

### Implementation Idea

Add the configuration for `maximumConcurrency` and modify Queue.ts. ( e.g. after [this line](https://github.com/getlift/lift/blob/master/src/constructs/aws/Queue.ts#L305))

```
this.configuration.worker.events = [
            // Subscribe the worker to the SQS queue
            {
                sqs: {
                    arn: this.queue.queueArn,
                    batchSize: batchSize,
                    maximumBatchingWindow: maximumBatchingWindow,
                    maximumConcurrency: maximumConcurrency
                    functionResponseType: "ReportBatchItemFailures",
                },
            },
        ];
```

**Comment by mnapoli** [comment]
That is 100% a good idea, thanks for opening this!

I'm wondering if `maximumConcurrency` shouldn't go one indentation lower, not in the `worker` but on the "queue" configuration itself, what do you think?

**Comment by htxiong** [comment]
Yaaas, I will work on a PR.

**Comment by htxiong** [comment]
@mnapoli can you review this PR for me?

**Comment by fredericbarthelet** [comment]
Released in [1.25.0](https://github.com/getlift/lift/releases/tag/1.25.0)
